src/typescript-service.ts

```diff
@@ -1050,6 +1038,49 @@ export class TypeScriptService {
 	}
 
 	/**
+	 * The completionItem/resolve request is used to fill in additional details from an incomplete
+	 * CompletionItem returned from the textDocument/completions call.
+	 *
+	 * @return Observable of JSON Patches that build a `CompletionItem` result
+	 */
+	completionItemResolve(item: CompletionItem, span = new Span()): Observable<Operation> {
+		const {uri, offset, entryName} = item.data;
+		const fileName: string = uri2path(uri);
+		return this.projectManager.ensureReferencedFiles(uri, undefined, undefined, span)
+			.toArray()
+			.map(() => {
+
+				const configuration = this.projectManager.getConfiguration(fileName);
+				configuration.ensureBasicFiles(span);
+
+				const details = configuration.getService().getCompletionEntryDetails(fileName, offset, entryName);
+				if (details) {
+					item.documentation = ts.displayPartsToString(details.documentation);
+					item.detail = ts.displayPartsToString(details.displayParts);
+					if (this.supportsCompletionWithSnippets) {
+						item.insertTextFormat = InsertTextFormat.Snippet;
+						if (details.kind === 'method' || details.kind === 'function') {
+							const parameters = details.displayParts
+								.filter(p => p.kind === 'parameterName')
+								.map((p, i) => '${' + `${i + 1}:${p.text}` + '}');
+							const paramString = parameters.join(', ');
+							item.insertText = details.name + `(${paramString})`;
+						} else {
+							item.insertText = details.name;
+
+						}
+					} else {
+						item.insertTextFormat = InsertTextFormat.PlainText;
+						item.insertText = details.name;
+					}
+					delete item.data;
```

Comment:
`delete` is slow, is this needed?

Reply:
It's not needed. It would be nice to reduce payload and not have to include the field in the asserts on resolved completion items (I wonder if the tests should even care at all what is in the data field?). Could we set to `undefined`, `null`, or should I just leave it be?